src/couch/src/couch_debug.erl

```diff
@@ -794,6 +795,68 @@ id("couch_file:init" ++ _, Pid, _Props) ->
 id(_IdStr, _Pid, _Props) ->
     "".
 
+first_call(Pid) ->
+    IC =
+        case process_info(Pid, initial_call) of
+            {initial_call, IC0} -> IC0;
+            undefined -> undefined
+        end,
+    Dict =
+        case process_info(Pid, dictionary) of
+            {dictionary, Dict0} -> Dict0;
+            undefined -> []
+        end,
+    MaybeCall = proplists:get_value('$initial_call', Dict, IC),
+    proplists:get_value(initial_call, Dict, MaybeCall).
+
+find_by_first_call(Module, Function) ->
+    FilterPids = fun(Pid) ->
+        case first_call(Pid) of
+            {Module, Function, _} -> true;
+            _ -> false
+        end
+    end,
+    lists:filter(FilterPids, processes()).
+
+view_report() ->
```

Comment:
I don't think this is the best place for this function. It knows too much about index internals. Let's move it into https://github.com/apache/couchdb/blob/3.x/src/couch_mrview/src/couch_mrview_debug.erl

Also I think the API should be:

- `processes() -> [pid()]`
- `processes_info(Pid) -> {pid(), id(), map()}.` - not sure about id()
- `report(Pid) -> prints report` - you would call the function above to get all info and format it nicelly

Reply:
The API will be `view_report(Pid) -> Prints report`. The first couple of lines in the function will be removed. They are solely for quick debugging right now.